Course Catalog

ZRUILIU opened the pull request:
View all available courses
Filter courses by category
Filter by price range
Sort by various criteria
See course previews
Read course descriptions and requirements
View instructor profiles

ZRUILIU committed:
Add student course viewing related functions
Implement student course list and details API routing
Create student course list and details page
Update dashboard page to display student course related statistics

ZRUILIU committed:
Add course browsing page and API routing
Create course API routing, support paging and category filtering
Add course browsing page, including category filtering and paging functions
Add course browsing entry on user course page

ZRUILIU committed:
Merge branch 'main' into Course-Page-Implementation

ZRUILIU committed:
Add price filtering and sorting functions
Implement price range filtering and multiple sorting options (latest, oldest, high and low prices, etc.)
Add price filtering input box and sorting drop-down menu on the front page
Add a button to clear all filters

ZRUILIU committed:
Add course details page
Add instructor details page

ZRUILIU committed:
yarn format

RuntaoZhuge commented on src/app/dashboard/courses/page.tsx:
```diff
@@ -0,0 +1,174 @@
+  );
+}
+
+function CourseCard({ course }: { course: Course }) {
```
This should be a component

ZRUILIU replied:
fixed

RuntaoZhuge commented on src/app/dashboard/instructor/[id]/page.tsx:
```diff
@@ -0,0 +1,208 @@
+import { useSession } from 'next-auth/react';
+import { UsersIcon, BookOpenIcon, ClockIcon } from '@/components/ui/Icons';
+
+interface User {
```
These should be placed under types/

ZRUILIU replied:
fixed

RuntaoZhuge commented on src/app/dashboard/instructor/[id]/page.tsx:
```diff
@@ -0,0 +1,208 @@
+    );
+  }
+
+  if (!instructor) {
```
shuold be named as isInstructor for any bools

ZRUILIU replied:
After looking at this code, I think there's a misunderstanding in the comments. The instructor variable isn't a boolean; it's of type `User | null` that stores the instructor's complete user data. The line `if (instructor)` is checking if the instructor data exists (not null), not checking a boolean flag.

ZRUILIU committed:
types: Extend the User interface and add the Course interface
Extract the CourseCard component to a separate file

ZRUILIU committed:
yarn format

ZRUILIU committed:
Fix the code according to the Deploy log

ZRUILIU committed:
Fix the code according to the Deploy log

ZRUILIU committed:
Split the core functions of the course browsing page into different components, including: course cards, filters, and paging components

All relevant interface definitions are moved to the src/types/ directory and are reasonably classified and organized by function.
